### ISSUE TITLE
Minimization - Inconsistent NCP calculation

### ISSUE THREAD

**Issue by abigailgold** [IBM/ai-privacy-toolkit#39]
NCP score is calculated on overall generalizations (combined from all cells), this includes when looking for a feature to remove from generalization (also taken from overall generalizations). 
Accuracy is computed on the representative values taken from cells. There can be a mismatch (i.e., generalization applied at cell level but not at overall level) which can lead to not achieving the target accuracy. Need separate NCP calculation for this purpose.

**Comment by abigailgold** [comment]
Another issue is that currently ncp is computed only once in the fit method. It would be beneficial to recalculate ncp in the transform method on the data that was actually transformed, or even to let the user send whatever data they wish to calculate ncp on.
I propose to make an external method for computing ncp that receives a (optional) dataset on which to compute and a pram indicating whether to compute it on the global generalizations or cell-based generalization.

**Comment by abigailgold** [comment]
Would also be best to be able to indicate to the fit() method how to apply generalizations in order to measure accuracy.
This way we have all options available: both NCP and accuracy can be measured either on global generalizations or on transformed data.

**Comment by abigailgold** [comment]
Also need to understand and resolve the difference between transform and _generalize methods.
